src/test/java/uk/gov/hmcts/dts/fact/services/validation/PostcodeValidatorTest.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.hmcts.dts.fact.services.validation;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.dts.fact.mapit.MapitData;
+import uk.gov.hmcts.dts.fact.services.MapitService;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = PostcodeValidator.class)
+public class PostcodeValidatorTest {
+
+    @MockBean
+    private MapitService mockMapitService;
+
+    @Autowired
+    private PostcodeValidator mapitValidator;
+
+    @BeforeEach
+    public void beforeEach() {
+        when(mockMapitService.getMapitData(Mockito.anyString()))
+            .thenReturn(Optional.of(new MapitData()));
+        when(mockMapitService.getMapitDataWithPartial(Mockito.anyString()))
+            .thenReturn(Optional.of(new MapitData()));
+    }
+
+    @Test
+    public void testIfPostcodesAreInvalid() {
+        for (String testPostcode : new String[]{"M", "", "P$11&PY", ".@£*$&,)@*&!@£,",
+            "ReEeEeeEaAaAllLlYLlLlLoOoOnNnG", "MM", "M1111", "M223J"}) {
+            assertFalse(mapitValidator.postcodeDataExists(testPostcode));
+        }
+    }
```

Comment:
Don't think you need to change that if the tests pass. I will approve the PR and up to you. What I mean in the previous comment is for tests which have a range of test cases, normally we will use ParameterizedTest in the following way, i.e. passing in a list of strings in ValueSource. JUnit will convert that into individual test cases for you.

What you have done before was passing in a single comma separated string and then parse that string yourself into an array for individual cases :)

@ParameterizedTest
@ValueSource(strings ={"M", "", "P$11&amp;PY", ".@£*$&amp;,)@*&amp;!@£,",            "ReEeEeeEaAaAllLlYLlLlLoOoOnNnG", "MM", "M1111", "M223J"})
public void testIfPostcodesAreInvalid() {
    assertFalse(mapitValidator.postcodeDataExists(testPostcode));
}